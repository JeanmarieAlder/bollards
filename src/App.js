import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from './Header';
import DetailsPage from './pages/DetailsPage';
import HomePage from './pages/HomePage';
import MapPage from './pages/MapPage';

function App() {
    return (
        <div className='main-div'>
            <Header></Header>
            <main role="main" className="container">
                <Switch>
                    <Route path='/map' component={MapPage} />
                    <Route path='/details/:bollard_id' component={DetailsPage} />
                    <Route path={['/home', '/list', '/']} component={HomePage} />
                </Switch>
            </main>
            
        </div>
    );
}

export default App;
