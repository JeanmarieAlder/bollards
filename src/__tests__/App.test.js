import React from "react";
import { render } from '@testing-library/react'
import App from "../App";
import { MemoryRouter, BrowserRouter } from "react-router-dom";

test("App displays correctly", () => {
    const component = render(
        <MemoryRouter>
            <App/>
        </MemoryRouter>
    );
    let tree = component.asFragment();
    console.log(tree)
    expect(tree).toMatchSnapshot();
})

test("App displays correctly with map route", () => {
    const component = render(
        <MemoryRouter initialEntries={["/map"]}>
            <App/>
        </MemoryRouter>
    );
    let tree = component.asFragment();
    console.log(tree)
    expect(tree).toMatchSnapshot();
})

test("App displays correctly with details route", () => {
    const component = render(
        <MemoryRouter initialEntries={["/details"]}>
            <App/>
        </MemoryRouter>
    );
    let tree = component.asFragment();
    console.log(tree)
    expect(tree).toMatchSnapshot();
})