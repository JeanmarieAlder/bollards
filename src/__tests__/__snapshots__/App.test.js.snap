// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App displays correctly 1`] = `
<DocumentFragment>
  <div
    class="main-div"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light fixed-top"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/home"
        >
          bollards.ch
        </a>
        <a
          class="skip-nav-link"
          href="#main-content"
        >
          s_skip_nav
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/list"
              href="/list"
            >
              s_list
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/map"
              href="/map"
            >
              s_map
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/news"
              href="/news"
            >
              s_news
            </a>
          </div>
          <div
            class="justify-content-end navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/about"
              href="/about"
            >
              s_about
            </a>
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                class="dropdown-toggle nav-link"
                href="#"
                id="basic-nav-dropdown"
                role="button"
                tabindex="0"
              >
                Français
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="container"
      id="main-content"
      role="main"
    >
      <div
        class="home-page-div"
      >
        <div
          id="home-page-title-div"
        >
          <h1
            class="title-text"
          >
            Bollards.ch
          </h1>
        </div>
        <div
          class="map-link-div"
        >
          <a
            href="/map"
          >
            <img
              alt="map-link"
              class="map-link-image"
              src="map_link.jpg"
            />
          </a>
        </div>
        <div
          class="search-box-div"
        >
          <input
            placeholder="Search Bollards"
            type="search"
          />
        </div>
        <div
          class="p-2 card-container row text-center justify-content-center"
        />
      </div>
    </main>
    <div
      class="cookie-consent-div"
    >
      <div
        class="CookieConsent"
        style="align-items: baseline; background: rgb(225, 237, 227); color: rgb(0, 0, 0); display: flex; flex-wrap: wrap; justify-content: space-between; left: 0px; position: fixed; width: 100%; z-index: 999; bottom: 0px;"
      >
        <div
          class=""
          style="flex: 1 0 300px; margin: 15px;"
        >
          s_cookies_policy_banner_text
          <button
            class="btn btn-link"
            type="button"
          >
            s_cookies_policy
          </button>
        </div>
        <div
          class=""
        >
          <button
            aria-label="Accept cookies"
            class=""
            id="rcc-confirm-button"
            style="background: rgb(255, 212, 45); border: 0px; border-radius: 0px; box-shadow: none; color: rgb(0, 0, 0); cursor: pointer; flex: 0 0 auto; padding: 5px 10px; margin: 15px; font-size: 13px;"
          >
            I agree.
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App displays correctly with details route 1`] = `
<DocumentFragment>
  <div
    class="main-div"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/home"
        >
          bollards.ch
        </a>
        <a
          class="skip-nav-link"
          href="#main-content"
        >
          s_skip_nav
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/list"
              href="/list"
            >
              s_list
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/map"
              href="/map"
            >
              s_map
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/news"
              href="/news"
            >
              s_news
            </a>
          </div>
          <div
            class="justify-content-end navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/about"
              href="/about"
            >
              s_about
            </a>
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                class="dropdown-toggle nav-link"
                href="#"
                id="basic-nav-dropdown"
                role="button"
                tabindex="0"
              >
                Français
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="container"
      id="main-content"
      role="main"
    />
    <div
      class="cookie-consent-div"
    >
      <div
        class="CookieConsent"
        style="align-items: baseline; background: rgb(225, 237, 227); color: rgb(0, 0, 0); display: flex; flex-wrap: wrap; justify-content: space-between; left: 0px; position: fixed; width: 100%; z-index: 999; bottom: 0px;"
      >
        <div
          class=""
          style="flex: 1 0 300px; margin: 15px;"
        >
          s_cookies_policy_banner_text
          <button
            class="btn btn-link"
            type="button"
          >
            s_cookies_policy
          </button>
        </div>
        <div
          class=""
        >
          <button
            aria-label="Accept cookies"
            class=""
            id="rcc-confirm-button"
            style="background: rgb(255, 212, 45); border: 0px; border-radius: 0px; box-shadow: none; color: rgb(0, 0, 0); cursor: pointer; flex: 0 0 auto; padding: 5px 10px; margin: 15px; font-size: 13px;"
          >
            I agree.
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App displays correctly with map route 1`] = `
<DocumentFragment>
  <div
    class="main-div"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/home"
        >
          bollards.ch
        </a>
        <a
          class="skip-nav-link"
          href="#main-content"
        >
          s_skip_nav
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/list"
              href="/list"
            >
              s_list
            </a>
            <a
              class="active nav-link"
              data-rr-ui-event-key="/map"
              href="/map"
            >
              s_map
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/news"
              href="/news"
            >
              s_news
            </a>
          </div>
          <div
            class="justify-content-end navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/about"
              href="/about"
            >
              s_about
            </a>
            <div
              class="nav-item dropdown"
            >
              <a
                aria-expanded="false"
                class="dropdown-toggle nav-link"
                href="#"
                id="basic-nav-dropdown"
                role="button"
                tabindex="0"
              >
                Français
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <main
      class="container"
      id="main-content"
      role="main"
    >
      <div>
        <div
          class="map-page-div leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
          style="position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            >
              <div
                class="leaflet-layer "
                style="z-index: 2;"
              >
                <div
                  class="leaflet-tile-container leaflet-zoom-animated"
                  style="z-index: 18; left: 0px; top: 0px;"
                >
                  <img
                    alt=""
                    class="leaflet-tile"
                    src="https://wmts20.geo.admin.ch/1.0.0/ch.swisstopo.swissimage/default/current/3857/10/529/361.jpeg"
                    style="width: 256px; height: 256px; left: -206px; top: -227px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-shadow-pane"
            />
            <div
              class="leaflet-pane leaflet-marker-pane"
            />
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            >
              <div
                class="leaflet-control-zoom leaflet-bar leaflet-control"
              >
                <a
                  aria-disabled="false"
                  aria-label="Zoom in"
                  class="leaflet-control-zoom-in"
                  href="#"
                  role="button"
                  title="Zoom in"
                >
                  <span
                    aria-hidden="true"
                  >
                    +
                  </span>
                </a>
                <a
                  aria-disabled="false"
                  aria-label="Zoom out"
                  class="leaflet-control-zoom-out"
                  href="#"
                  role="button"
                  title="Zoom out"
                >
                  <span
                    aria-hidden="true"
                  >
                    −
                  </span>
                </a>
              </div>
            </div>
            <div
              class="leaflet-top leaflet-right"
            >
              <div
                aria-haspopup="true"
                class="leaflet-control-layers leaflet-control-layers-expanded leaflet-control"
              >
                <a
                  class="leaflet-control-layers-toggle"
                  href="#"
                  role="button"
                  title="Layers"
                />
                <section
                  class="leaflet-control-layers-list leaflet-control-layers-scrollbar"
                  style="height: -50px;"
                >
                  <div
                    class="leaflet-control-layers-base"
                  >
                    <label>
                      <span>
                        <input
                          class="leaflet-control-layers-selector"
                          name="leaflet-base-layers_28"
                          type="radio"
                        />
                        <span>
                           s_map
                        </span>
                      </span>
                    </label>
                    <label>
                      <span>
                        <input
                          checked="checked"
                          class="leaflet-control-layers-selector"
                          name="leaflet-base-layers_28"
                          type="radio"
                        />
                        <span>
                           s_satellite
                        </span>
                      </span>
                    </label>
                  </div>
                  <div
                    class="leaflet-control-layers-separator"
                    style="display: none;"
                  />
                  <div
                    class="leaflet-control-layers-overlays"
                  />
                </section>
              </div>
            </div>
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JavaScript library for interactive maps"
                >
                  Leaflet
                </a>
                 
                <span
                  aria-hidden="true"
                >
                  |
                </span>
                 © Data: 
                <a
                  href="https://www.swisstopo.admin.ch/en/home.html"
                >
                  CNES, Spot Image, swisstopo, NPOC
                </a>
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
    </main>
    <div
      class="cookie-consent-div"
    >
      <div
        class="CookieConsent"
        style="align-items: baseline; background: rgb(225, 237, 227); color: rgb(0, 0, 0); display: flex; flex-wrap: wrap; justify-content: space-between; left: 0px; position: fixed; width: 100%; z-index: 999; bottom: 0px;"
      >
        <div
          class=""
          style="flex: 1 0 300px; margin: 15px;"
        >
          s_cookies_policy_banner_text
          <button
            class="btn btn-link"
            type="button"
          >
            s_cookies_policy
          </button>
        </div>
        <div
          class=""
        >
          <button
            aria-label="Accept cookies"
            class=""
            id="rcc-confirm-button"
            style="background: rgb(255, 212, 45); border: 0px; border-radius: 0px; box-shadow: none; color: rgb(0, 0, 0); cursor: pointer; flex: 0 0 auto; padding: 5px 10px; margin: 15px; font-size: 13px;"
          >
            I agree.
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
